paasta_tools/cli/cmds/local_run.py

```diff
@@ -653,6 +690,98 @@ def check_if_port_free(port):
+            file=sys.stderr,
+        )
+        sys.exit(1)
+    try:
+        with open("/nail/etc/runtimeenv") as runtimeenv_file:
+            aws_account = runtimeenv_file.read()
+            # Map runtimeenv in special cases to proper aws account name
+            if aws_account == "stage":
+                aws_account = "dev"
+            elif aws_account == "corp":
+                aws_account = "corpprod"
+    except FileNotFoundError:
+        print(
+            "Unable to determine environment for AWS account name. Using 'dev'",
+            file=sys.stderr,
+        )
+        aws_account = "dev"
+    if pod_identity and (assume_pod_identity or assume_role_arn):
+        print(
+            "Calling aws-okta to assume role {} using account {}".format(
+                pod_identity, aws_account
+            )
+        )
+    elif use_okta_role:
+        print(f"Calling aws-okta using account {aws_account}")
+    else:
+        # use_okta_role, assume_pod_identity, and assume_role are all empty. This shouldn't happen
+        print(
+            "Error: assume_aws_role called without required arguments", file=sys.stderr
+        )
+        sys.exit(1)
+    # local-run will sometimes run as root - make sure that we get the actual
+    # users AWS credentials instead of looking for non-existent root AWS
+    # credentials
+    if os.getuid() == 0:
+        aws_okta_cmd = [
+            "sudo",
+            "-u",
+            get_username(),
+            f"HOME=/nail/home/{get_username()}",
+            "aws-okta",
+            "-a",
+            aws_account,
+            "-o",
+            "json",
+        ]
+    else:
+        aws_okta_cmd = ["aws-okta", "-a", aws_account, "-o", "json"]
+    cmd = subprocess.run(aws_okta_cmd, stdout=subprocess.PIPE)
+    if cmd.returncode != 0:
+        print(
+            "Error calling aws-okta. Remove --assume-pod-identity to run without pod identity role",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+    cmd_output = json.loads(cmd.stdout.decode("utf-8"))
+
+    if not use_okta_role:
+        boto_session = boto3.Session(
+            aws_access_key_id=cmd_output["AccessKeyId"],
+            aws_secret_access_key=cmd_output["SecretAccessKey"],
+            aws_session_token=cmd_output["SessionToken"],
+        )
+        sts_client = boto_session.client("sts")
+        assumed_role = sts_client.assume_role(
+            RoleArn=pod_identity, RoleSessionName=f"{get_username()}-local-run"
+        )
+        # The contents of "Credentials" key from assume_role is the same as from aws-okta
+        cmd_output = assumed_role["Credentials"]
+
+    creds_dict = {
+        "AWS_ACCESS_KEY_ID": cmd_output["AccessKeyId"],
+        "AWS_SECRET_ACCESS_KEY": cmd_output["SecretAccessKey"],
+        "AWS_SESSION_TOKEN": cmd_output["SessionToken"],
+    }
+    return creds_dict
```

Comment:
looks like mypy isn't happy with this for some reason?
I'm seeing: `paasta_tools/cli/cmds/local_run.py:782: error: Incompatible return value type (got "Dict[str, Any]", expected "AWSSessionCreds")` in the tests

i'm not quite sure why it's not able to infer things here (maybe our mypy is too old?) - but something like 
```suggestion
    creds_dict: AWSSessionCreds = {
        "AWS_ACCESS_KEY_ID": cmd_output["AccessKeyId"],
        "AWS_SECRET_ACCESS_KEY": cmd_output["SecretAccessKey"],
        "AWS_SESSION_TOKEN": cmd_output["SessionToken"],
    }
    return creds_dict
```
or 
```suggestion
    creds_dict = {
        "AWS_ACCESS_KEY_ID": cmd_output["AccessKeyId"],
        "AWS_SECRET_ACCESS_KEY": cmd_output["SecretAccessKey"],
        "AWS_SESSION_TOKEN": cmd_output["SessionToken"],
    }
    return cast(AWSSessionCreds, creds_dict)
```
*should* resolve that